app/models/subnet.rb

```diff
@@ -396,6 +409,13 @@ def ensure_ip_addrs_valid
     end
   end
 
+  def ensure_gateway_in_subnet
+    masked_gw = IPAddr.new(gateway).mask(mask)
+    errors.add(:gateway, _("is not from the subnet")) unless masked_subnet_address == masked_gw
```

Comment:
why not:

```suggestion
    gw = IPAddr.new(gateway)
    errors.add(:gateway, _("is not from the subnet")) unless masked_subnet_address.include?(gw)
```

but then see above…

Reply:
Oh yeah your solution is definitely better than mine. Not sure what you mean "see above". Oh perhaps the below comment, we should definitely return the same type. I will fix this.

Reply:
Yeah, I meant the type comment, thanks GH for sorting it differently ;)